ocean_spark/hooks.py

```diff
@@ -92,37 +95,49 @@ def _do_api_call(
                 )
                 response.raise_for_status()
                 return response.json()
+            except (
+                requests_exceptions.ConnectionError,
+                requests_exceptions.Timeout,
+            ) as e:
+                self.log.error(
+                    "Attempt %d Request to Ocean Spark API failed with reason: %s",
+                    attempt_num,
+                    str(e),
+                )
             except requests_exceptions.RequestException as e:
-                if not _retryable_error(e):
-                    # In this case, the user probably made a mistake.
-                    # Don't retry.
-                    raise AirflowException(
-                        "Response: {0}, Status Code: {1}".format(
-                            e.response.content, e.response.status_code
-                        )
-                    )
-
-                self._log_request_error(attempt_num, e)
+                self._handle_request_exception(e)
 
             if attempt_num == self.retry_limit:
                 raise AirflowException(
-                    (
-                        "API requests to Data Mechanics failed {} times. "
-                        + "Giving up."
-                    ).format(self.retry_limit)
+                    f"Request to Ocean Spark API failed failed {self.retry_limit} times. Giving up."
                 )
 
             attempt_num += 1
             time.sleep(self.retry_delay.total_seconds())
 
-    def _log_request_error(
-        self, attempt_num: int, error: requests_exceptions.RequestException
+    def _handle_request_exception(
+        self, ex: requests_exceptions.RequestException
     ) -> None:
-        self.log.error(
-            "Attempt %s API Request to Data Mechanics failed with reason: %s",
-            attempt_num,
-            error,
-        )
+        try:
+            api_response: ApiResponse = json.loads(ex.response.content)
+            request_id = api_response["request"]["id"]
+            errors = api_response["response"]["errors"]
+            status_code = api_response["response"]["status"]["code"]
+            self.log.error(
+                "Request %s, to Ocean Spark API failed with errors: %s",
+                request_id,
+                errors,
+            )
+            raise AirflowException(
+                f"RequestID: {request_id}, errors: {errors}, Status Code: {status_code}"
+            )
```

Comment:
I'm not a huge fan of the `log-rethrow` pattern being used here.

I think I might just remove the log line and relyon the calling code to log the exception properly. As seem  to be the case in airflow anyway.

Reply:
Tried this approach and it does not work.